Add some explanation for the core concepts around channels (#1457)
feathersjs/docs: api/channels.md

- Only admins should be notified when new users are created
- When a user is created, modified or removed, non-admins should only receive a "safe" version of the user object (e.g. only `email`, `id` and `avatar`)

## Concepts

> When using channels, the server pushes events (such as "created", "removed" etc. for a particular service) down to its clients via *channels*. The client doesn’t listen to these channels directly, but rather subscribes to specific events (e.g. on services) that it is interested in. Those events will only fire if the server pushes relevant data down a channel that the client is connected to.

Reviewer: ```suggestion
When using channels, the server pushes events (such as "created", "removed" etc. for a particular service) down to its clients via *channels*. The client doesn’t listen to individual channels directly, but rather subscribes to specific events on services that it is interested in. Those events will only fire on the client if the server pushes data to one or more channels that the client has been added to.
```

Author: Agree with these changes but ... how do I actually accept them into the PR?
